### DISCUSSION TITLE
When suggesting Patterns to Autocomplete, use Expr to influence generation

### DISCUSSION THREAD

**Discussion by StachuDotNet** [darklang/dark#4464] [Suggestions]
If I'm writing the code:

```
match Ok "testing"
  | ~___ -> ___
```
(with my cursor at the `~`),

a limited standard set of Autocomplete Patterns are suggested. `true`, `false`, `OK ___`, etc.

Given that Dark has context of the `Ok "testing"` expression that we're matching against, it could be additionally useful to include `Ok "testing" as an option - the PString can be derived by the expr.

This is sort of contrived, but I suspect there are more cases where this sort of thing could be handy? Maybe not :)

**Comment by pbiggar** [comment]
I have been thinking to have a command that generates all possible cases for a particular type. Another way to do that would be like copilot: each line individually generated and accepted. In this case, we could just generate each possible sumtype from the type of the expression.
